v2/internal/controllers/crd_synapse_test.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"testing"
+
+	storage "github.com/Azure/azure-service-operator/v2/api/storage/v1beta20210401"
+	synapse "github.com/Azure/azure-service-operator/v2/api/synapse/v1api20210601"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/azure-service-operator/v2/internal/util/to"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	. "github.com/onsi/gomega"
+)
+
+func Test_Workspace_BigDataPool(t *testing.T) {
+	t.Parallel()
+	tc := globalTestContext.ForTest(t)
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	configMapName := "my-configmap"
+	accountUrlKey := "accountUrl"
+
+	// Create a storage account to use as workspace datalake storage
+	sa := &storage.StorageAccount{
+		ObjectMeta: tc.MakeObjectMetaWithName(tc.NoSpaceNamer.GenerateName("adlstore")),
+		Spec: storage.StorageAccount_Spec{
+			Location:     to.Ptr("eastus2"),
+			Owner:        testcommon.AsOwner(rg),
+			Kind:         to.Ptr(storage.StorageAccount_Kind_Spec_StorageV2),
+			Sku:          &storage.Sku{Name: to.Ptr(storage.SkuName_Standard_LRS)},
+			IsHnsEnabled: to.Ptr(true),
+			OperatorSpec: &storage.StorageAccountOperatorSpec{
+				ConfigMaps: &storage.StorageAccountOperatorConfigMaps{
```

Comment:
ConfigMaps is implemented in: https://github.com/Azure/azure-service-operator/pull/2855